File: 상현 알고리즘/모의고사.py
def solution(answers):
    one = [1,2,3,4,5,1,2,3,4,5]
    two = [2,1,2,3,2,4,2,5]
    three = [3,3,1,1,2,2,4,4,5,5]
    score = [0,0,0]
    result = []
    
    for index, value in enumerate(answers):
        if one[index%10] == value :
            score[0] +=1
        if two[index%8] == value :
            score[1] +=1
        if three[index%10] == value :
            score[2] +=1

    for i , v in enumerate(score):
        if v == max(score):
            result.append(i+1)
    return result

File: thyoondev/모의고사.js
function repeatStr(str, maxLength) {
  return str.repeat(maxLength).substr(0, maxLength);
}

function makePattern(str, maxLength) {
  return [...repeatStr(str, maxLength)].map(Number);
}

function solution(answers) {
  var answer = [];
  //찍는 패턴 배열로 만들기
  let one = makePattern('12345', answers.length);
  let two = makePattern('21232425', answers.length);
  let three = makePattern('3311224455', answers.length);

  //정답 수 체크
  let oneAnswersCount = answers.filter((e, i) => e === one[i]).length;
  let twoAnswersCount = answers.filter((e, i) => e === two[i]).length;
  let threeAnswersCount = answers.filter((e, i) => e === three[i]).length;
  let max = Math.max(oneAnswersCount, twoAnswersCount, threeAnswersCount);

  //정답 넣기
  if (oneAnswersCount === max) {
    answer.push(1);
  }
  if (twoAnswersCount === max) {
    answer.push(2);
  }
  if (threeAnswersCount === max) {
    answer.push(3);
  }

  return answer;
}
